### PR TITLE
Fix bugs in garage/tf/baselines and optimizers

### PR DESCRIPTION
**Pull request by CatherineSue**
There are some missing lines in garage/tf/optimizers. Without them, the
penalty_lbfgs_optimizer can't work properly.

Add baselines to garage/tf so we can use tf/regressors.

### CODE REVIEW

File: garage/tf/regressors/gaussian_mlp_regressor.py [hunk 1]
```diff
@@ -68,10 +69,14 @@ def __init__(self,
...
+                if optimizer_args is None:
+                    optimizer_args = dict()
                 if use_trust_region:
-                    optimizer = PenaltyLbfgsOptimizer("optimizer")
+                    optimizer = PenaltyLbfgsOptimizer(**optimizer_args)
                 else:
-                    optimizer = LbfgsOptimizer("optimizer")
+                    optimizer = LbfgsOptimizer(**optimizer_args)
+            else:
+                optimizer = optimizer(**optimizer_args)
```
Comment (gonzaiva) [inline garage/tf/regressors/gaussian_mlp_regressor.py]: Is optimizer_args different from None in this else branch?

File: garage/tf/optimizers/penalty_lbfgs_optimizer.py [hunk 2]
```diff
@@ -57,6 +57,10 @@ def update_opt(self,
         :param inputs: A list of symbolic variables as inputs
         :return: No return value.
         """
+        constraint_term, constraint_value = leq_constraint
+        penalty_var = tf.placeholder(tf.float32, tuple(), name="penalty")
+        penalized_loss = loss + penalty_var * constraint_term
```
Comment (gonzaiva) [inline garage/tf/optimizers/penalty_lbfgs_optimizer.py]: I see these variables are obtained again in the lines of code below, but under a scope. Why are they required outside the scope?

File: garage/tf/regressors/deterministic_mlp_regressor.py [hunk 3]
```diff
@@ -42,7 +44,11 @@ def __init__(
         with tf.variable_scope(name, "DeterministicMLPRegressor"):
 
             if optimizer is None:
-                optimizer = LbfgsOptimizer(name="optimizer")
+                if optimizer_args is None:
+                    optimizer_args = dict()
+                optimizer = LbfgsOptimizer(**optimizer_args)
+            else:
+                optimizer = optimizer(**optimizer_args)
```
Comment (gonzaiva) [inline garage/tf/regressors/deterministic_mlp_regressor.py]: Is optimizer_args different from None in this else branch?

File: garage/tf/optimizers/penalty_lbfgs_optimizer.py (hunk 2 above)
Comment (CatherineSue) [reply garage/tf/optimizers/penalty_lbfgs_optimizer.py]: Oh my bad. I didn't see they are obtained under a scope.

Comment (CatherineSue) [reply garage/tf/optimizers/penalty_lbfgs_optimizer.py]: But isn't `self._max_constraint_val = constraint_value` requires the line 60?

File: garage/tf/regressors/gaussian_mlp_regressor.py (hunk 1 above)
Comment (CatherineSue) [reply garage/tf/regressors/gaussian_mlp_regressor.py]: No. Why?

Comment (gonzaiva) [reply garage/tf/regressors/gaussian_mlp_regressor.py]: I saw you added this [guard](https://github.com/rlworkgroup/garage/blob/be570838f2fe4c5f291b10247209d04548565370/garage/tf/regressors/gaussian_mlp_regressor.py#L72-L73) for null args, so I was wondering if the same could happen when optimizer is already set, but if that's not the case, it's fine.

File: garage/tf/optimizers/penalty_lbfgs_optimizer.py (hunk 2 above)
Comment (gonzaiva) [reply garage/tf/optimizers/penalty_lbfgs_optimizer.py]: Oh you're right, and using git blame it seems I'm responsible for that. Could you please change the code to:
```
        params = target.get_params(trainable=True)
        with tf.name_scope(name, "PenaltyLbfgsOptimizer",
                           [leq_constraint, loss, params]):
            constraint_term, constraint_value = leq_constraint
            penalty_var = tf.placeholder(tf.float32, tuple(), name="penalty")
            penalized_loss = loss + penalty_var * constraint_term

            self._target = target
            self._max_constraint_val = constraint_value
            self._constraint_name = constraint_name

            def get_opt_output():
```
Sorry for the inconvenience.

**Pull request merged**